syntax = "proto3";

service NodeService {
  rpc processTransaction (TransactionRequest) returns (TransactionResponse);
  rpc nodeUpdate (NodeDataUpdate) returns (StatusResponse);
}

message TransactionRequest {
  string transaction = 1;
}

message TransactionResponse {
  int32 status = 1;
  string answer = 2;
}

message NodeDataUpdate {
  string sender = 1;
  string data = 2;
}

message StatusResponse {
  int32 status = 1;
  string status_message = 2;
}